### PR TITLE
feat(server,web): make user deletion delay configurable

### PR DESCRIPTION
**Pull request by samholton**
First part to add a new User Settings section to add configuration option for user deletion delay. Next up is to add a force option to queue up user for immediate deletion.

delete delay

### CODE REVIEW

File: web/src/lib/components/admin-page/settings/admin-settings.svelte [hunk 1]
```diff
@@ -35,6 +36,8 @@
       savedConfig = cloneDeep(newConfig);
       notificationController.show({ message: 'Settings saved', type: NotificationType.Info });
 
+      await loadConfig();
```
Comment (samholton) [inline web/src/lib/components/admin-page/settings/admin-settings.svelte:39]: Somewhat outside the scope of this PR, but added this to ensure the system settings are reloaded after they are saved. This is used in the confirmation screen when deleting a user. Without this change, updating the setting and then removing a user without reloading the page will still show the old value.

Comment (jrasm91) [reply web/src/lib/components/admin-page/settings/admin-settings.svelte:39]: newConfig should be the latest config. Maybe we just need to update the store with the new value.

File: web/src/lib/components/admin-page/settings/user-settings/user-settings.svelte [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+</script>
+
+<div>
+  <div in:fade={{ duration: 500 }}>
+    <form autocomplete="off" on:submit|preventDefault>
+      <div class="ml-4 mt-4 flex flex-col gap-4">
+        <SettingInputField
+          inputType={SettingInputFieldType.NUMBER}
+          label="DELETE DELAY"
+          desc="Number of days after removal to permanently delete an user's account and assets. Changing this will impact users already queued for deletion."
```
Comment (jrasm91) [inline web/src/lib/components/admin-page/settings/user-settings/user-settings.svelte:28]: I think you can add min=1 here

**Review by jrasm91 — COMMENTED** [review]
Looks good.

**Review by jrasm91 — APPROVED** [review]

File: web/src/lib/components/admin-page/settings/user-settings/user-settings.svelte (hunk 2 above)
Comment (samholton) [reply web/src/lib/components/admin-page/settings/user-settings/user-settings.svelte:29]: Thanks, just noticed I was missing that

**Comment by jrasm91** [comment]
I mean it might need to be min={1}

File: web/src/lib/components/admin-page/settings/admin-settings.svelte (hunk 1 above)
Comment (samholton) [reply web/src/lib/components/admin-page/settings/admin-settings.svelte:39]: Any examples of that being done somewhere? I tried replacing this with 
```
serverConfig.update(() => ({ ...savedConfig, loaded: true }));
``` 
Also tried newConfig but the delete confirm modal shows undefined after making a settings update and not refreshing page.

Comment (samholton) [reply web/src/lib/components/admin-page/settings/admin-settings.svelte:39]: Ah, its different, one is `SystemConfigDto` and the other is `ServerConfigDto`

Comment (samholton) [reply web/src/lib/components/admin-page/settings/admin-settings.svelte:39]: Ok, this works for my specific field and saves the network call. But does not refresh any other fields. What do you think, network call and ensure everything is up to date, or pick and choose what is updated?

```
serverConfig.update(() => ({ userDeleteDelay: newConfig.user.deleteDelay, loaded: true }));
```

**Comment by samholton** [comment]
> I mean it might need to be min={1}

`min=1` worked but updated it to be consistent with the others

**Comment by alextran1502** [comment]
Hello, thank you for the PR! What is the impact of changing the number of day for already queued for delete user?

**Comment by samholton** [comment]
> Hello, thank you for the PR! What is the impact of changing the number of day for already queued for delete user?

The number of days is not stored with the user record, so each time the batch job runs, it will evaluate the queued users against the current setting value. 

I attempted to indicate this in the text: https://github.com/immich-app/immich/pull/7663/files#diff-2a84dfc76e5b23ae7a77e3ba11ca749128214aff7a18e5e207acfe551df563fbR29

File: web/src/lib/components/admin-page/settings/admin-settings.svelte (hunk 1 above)
Comment (jrasm91) [reply web/src/lib/components/admin-page/settings/admin-settings.svelte:39]: Oh right, this is just the public config. You can leave the load config in for now. I think the server can emit an event through the websocket to push out the update though, but it doesn't really matter.

**Comment by jrasm91** [comment]
We should maybe clarify somewhere that a job runs at midnight to check for users that are ready to be deleted and then we can add that this change will take effect the next time the job runs.

**Comment by samholton** [comment]
updated

updated text

**Comment by jrasm91** [comment]
Yeah that looks good to me.

**Pull request merged**